distribution/systemd/vector.service

```diff
@@ -11,7 +13,8 @@ ExecStartPre=/usr/bin/vector validate
 ExecStart=/usr/bin/vector
 ExecReload=/usr/bin/vector validate
 ExecReload=/bin/kill -HUP $MAINPID
-Restart=no
+Restart=always
+RestartSec=10
 AmbientCapabilities=CAP_NET_BIND_SERVICE
 EnvironmentFile=-/etc/default/vector
 
```

Comment:
```suggestion
EnvironmentFile=-/etc/default/vector
# Since systemd 229, should be in [Unit] but in order to support systemd <229,
# it is also supported to have it here.
StartLimitInterval=10
StartLimitBurst=5
```

[Pulled from the DD Agent](https://github.com/DataDog/datadog-agent/blob/main/omnibus/config/templates/datadog-agent/systemd.service.erb#LL12-L15C18)